### PR TITLE
refactor(ast, regular_expression)!: converge `oxc_ast::RegExpFlags` and `oxc_regular_expression::ast::Flags`.

### LINKED ISSUE #5562: Use `bitflags` for `oxc_regular_expression::ast::Flags`
Currently our `oxc_ast::ast::literal::RegExpFlags` is defined using the `bitflags` macro.

https://github.com/oxc-project/oxc/blob/af69393a1c72f58606e01875d69f0bad85d646e7/crates/oxc_ast/src/ast/literal.rs#L156-L197

While the `oxc_regular_expression::ast::Flags` is defined as such:

https://github.com/oxc-project/oxc/blob/af69393a1c72f58606e01875d69f0bad85d646e7/crates/oxc_regular_expression/src/ast.rs#L26-L40

I suggest we refactor `Flags` to use `bitflags` and maybe rename it to `RegularExpressionFlags`. Ideally, we should remove the `RegExpFlags` and share this flag and its parser(probably through `RegularExpressionFlags::try_from("giv")`) throughout the project.

We can still keep a `FlagsOption` that implements `Into<RegularExpressionFlags>` if we want to allow users to use a struct to pass in flags to the pattern parser.

cc @leaysgur @overlookmotel

### PR DESCRIPTION
**Pull request by rzvxa**
closes #5562

### CODE REVIEW

**Comment by rzvxa** [comment]
* **#5592** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5592?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_regular_expression/src/ast.rs [hunk 1]
```diff
@@ -360,3 +356,98 @@ pub struct NamedReference<'a> {
...
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+#[cfg(feature = "serialize")]
+#[wasm_bindgen::prelude::wasm_bindgen(typescript_custom_section)]
+const TS_APPEND_CONTENT: &'static str = r#"
```
Comment (rzvxa) [inline crates/oxc_regular_expression/src/ast.rs:434]: Should I change any javascript code now that this binding is renamed?

**Review by Boshen — APPROVED** [review]

File: crates/oxc_regular_expression/src/options.rs [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
-        ParserOptions { unicode_mode: true, ..self }
+    pub fn with_span_offset(self, span_offset: u32) -> Self {
+        Self { span_offset, ..self }
     }
 
     #[must_use]
-    pub fn with_unicode_sets_mode(self) -> ParserOptions {
-        ParserOptions { unicode_mode: true, unicode_sets_mode: true, ..self }
+    pub fn with_flags(self, flags: RegularExpressionFlags) -> Self {
+        Self { flags, ..self }
```
Comment (leaysgur) [inline crates/oxc_regular_expression/src/options.rs:32]: I prefer this in order to split the domain more explicit. 👀 

```suggestion
        Self {
            unicode_mode: flags.intersects(RegularExpressionFlags::U | RegularExpressionFlags::V),
            unicode_sets_mode: flags.contains(RegularExpressionFlags::V),
            ..self
        }
```

**Comment by leaysgur** [comment]
Closing, `oxc_regular_expression::ast::Flags` is removed in #6262

**Pull request closed without merging**